inc/commondbchild.class.php

```diff
@@ -86,6 +89,48 @@ static function getSQLRequestToSearchForItem($itemtype, $items_id) {
    }
 
 
+   /**
+    * Get request cirteria to search for an item
+    *
+    * @since 9.4
+    *
+    * @param string  $itemtype Item type
+    * @param integer $items_id Item ID
+    *
+    * @return array|null
+   **/
+   static function getSQLCriteriaToSearchForItem($itemtype, $items_id) {
+
+      $criteria = [
+         'SELECT' => [
+            static::getIndexName(),
+            static::$items_id . ' AS items_id',
+            new \QueryExpression("'" . static::$itemtype . "' AS itemtype")
```

Comment:
Will not for when `static::$itemtype` equls to `itemtype`. In this case, we want to query the `itemtype field value`.

Reply:
```php
if (preg_match('/^itemtype/', static::$itemtype)) {
   $criteria['SELECT'][] = 'itemtype';
} else {
   $criteria['SELECT'][] = new \QueryExpression("'" . static::$itemtype . "' AS itemtype");
}
```